cmd/root.go

```diff
@@ -66,16 +66,20 @@ func InitProfile(cmd *cobra.Command, onlyNew bool) *profiles.ProfileStore {
 // TODO make this a preRun hook
 func NewHandler(cmd *cobra.Command) handlers.Handler {
 	fh := cli.NewFlagHelper(cmd)
+
+	// Non-profile flags
 	host := fh.GetOptionalString("host")
 	tlsNoVerify := fh.GetOptionalBool("tls-no-verify")
 	withClientCreds := fh.GetOptionalString("with-client-creds")
 	withClientCredsFile := fh.GetOptionalString("with-client-creds-file")
 
 	// if global flags are set then validate and create a temporary profile in memory
 	var cp *profiles.ProfileStore
-	if host != "" || withClientCreds != "" || withClientCredsFile != "" {
-		err := errors.New("when using global flags --host, --with-client-creds, or --with-client-creds-file, " +
-			"profiles will not be used and all required flags must be set")
+	if host != "" || tlsNoVerify || withClientCreds != "" || withClientCredsFile != "" {
```

Comment:
should `tlsNoVerify` on any command always trigger use of an in memory profile?

Reply:
I would say yes, if you need to ignore TLS verification then you need to specify it upon creation. Otherwise, it will lead to bad habits that make users prone to MITM attacks.

User's might assume that `--tls-no-verify` is an escape hatch and leak their app id or client I'd to a MITM.